Hi @alonfnt, Thanks so much for your pull request! I tried to test your branch with Python 3.12 but faced the following error when I tried to load a pickle file:

```
Traceback (most recent call last):
  File "/workspace/CellModeller/CellModeller/GUI/PyGLCMViewer.py", line 194, in loadPickle
    sim = Simulator(self.modName, \
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/CellModeller/CellModeller/Simulator.py", line 80, in __init__
    self.module = importlib.util.import_from_spec(module_spec)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'importlib.util' has no attribute 'import_from_spec'
Aborted (core dumped)
```

When I looked into the change I found that we should be using `module_from_spec` instead of `import_from_spec` in [this line](https://github.com/cellmodeller/CellModeller/blob/4da7a901c7a34230094f2f971380151ba946a068/CellModeller/Simulator.py#L80) . When I made the new change, I was able to load the pickle file successfully and resume the simulation.

Please make this change in your branch so we can merge it soon. Thanks again.

--- a/CellModeller/Regulation/ModuleRegulator.py
+++ b/CellModeller/Regulation/ModuleRegulator.py
@@ -1,7 +1,6 @@
 import copy
 import os.path
 import sys
-import imp
 
 class ModuleRegulator:
     def __init__(self, sim, biophys=None, signalling=None):

--- a/CellModeller/Simulator.py
+++ b/CellModeller/Simulator.py
@@ -7,7 +7,6 @@ import pickle
 import csv
 import numpy
 import inspect
-import imp
 import configparser
 import importlib
 
@@ -77,7 +76,8 @@ visualised.
         self.moduleStr = moduleStr # Import stored python code string (from a pickle usually)
         if self.moduleStr:
             print("Importing model %s from string"%(self.moduleName))
-            self.module = imp.new_module(moduleName)
+            module_spec = importlib.util.spec_from_loader(moduleName, loader=None)
+            self.module = importlib.util.module_from_spec(module_spec)
             exec(moduleStr, self.module.__dict__)
         else:
             # In case the moduleName is a path to a python file:

